### PR TITLE
feat(rpc profile events): Implement enhanced RPC endpoint functionality with tracing and profiling

### PR DESCRIPTION
**Pull request by nachivrn**
This PR introduces comprehensive improvements to RPC endpoint handling,
adding support for debug modes, trace visualization, Query information and profile event analysis. 

- Add new EndpointSelector component for RPC endpoint selection
- Implement ExampleRequestAccordion for request payload management
- Create RequestInput component for handling user inputs and debug modes
- Add ResponseDisplay component for visualizing query results
- Implement MetadataTable and ProfileEventsTable components
- Add SummarizedTraceDisplay for trace visualization
- Update types and utilities for trace/profile processing
- Refactor accordion styles for improved UI organization
- Added summarizeTraceWithProfile method to Client interface
- Updated types.ts with new interfaces for query summaries and profile events
- Refactored clickhouse_trace_query to separate profile event gathering
- The abort controller pattern is implemented using React's useRef hook to maintain a reference to the current AbortController instance, which can be used to cancel pending requests.

Screenshot 2024-11-04 at 10 29 24 AM
Screenshot 2024-11-04 at 10 29 46 AM
Screenshot 2024-11-04 at 10 34 55 AM

### CODE REVIEW

**Review by evanh — COMMENTED** [review]
This looks good to me, love how you separated out the UI components.

File: snuba/admin/views.py [hunk 1]
```diff
@@ -571,6 +531,56 @@ def clickhouse_trace_query() -> Response:
         )
 
 
+def gather_profile_events(query_trace: TraceOutput, storage: str) -> None:
```
Comment (volokluev) [inline snuba/admin/views.py:534]: this file is pretty heavily overloaded, can you move this into its own file?

File: snuba/admin/views.py [hunk 2]
```diff
@@ -571,6 +531,56 @@ def clickhouse_trace_query() -> Response:
...
+            system_query_result = run_system_query_on_host_with_sql(
+                query_trace_data.host,
+                int(query_trace_data.port),
+                storage,
+                sql,
+                False,
+                g.user,
+            )
+            if not system_query_result.results:
+                time.sleep(1)
```
Comment (volokluev) [inline snuba/admin/views.py:563]: you could probably do exponential backoff rather than polling every second

**Review by volokluev — APPROVED** [review]
changes overall LGTM, just some minor adjustments to make

File: snuba/admin/views.py (hunk 1 above)
Comment (nachivrn) [reply snuba/admin/views.py:534]: Yes. I will refactor the changes related to gather_profile_events in a separate follow-up PR

File: snuba/admin/views.py [hunk 3]
```diff
@@ -571,6 +531,56 @@ def clickhouse_trace_query() -> Response:
...
+    profile_events_raw_sql = "SELECT ProfileEvents FROM system.query_log WHERE query_id = '{}' AND type = 'QueryFinish'"
+    for query_trace_data in parse_trace_for_query_ids(query_trace):
+        sql = profile_events_raw_sql.format(query_trace_data.query_id)
+        logger.info(
+            "Gathering profile event using host: {}, port = {}, storage = {}, sql = {}, g.user = {}".format(
+                query_trace_data.host, query_trace_data.port, storage, sql, g.user
+            )
+        )
+        system_query_result, counter = None, 0
+        while counter < 30:
```
[the remark this answers is not in the document]
Comment (nachivrn) [reply snuba/admin/views.py:551]: Yes. I will refactor the changes related to gather_profile_events in a separate follow-up PR

**Pull request merged**